content/6_Advanced/String_Search.mdx

```diff
@@ -139,6 +138,66 @@ but instead finds the longest common prefix of a string and all of its suffixes.
 	<Resource source="CF" title="Z Algorithm" url="3107" />
 </Resources>
 
+### Explanation
+
+The Z-algorithm is very similar to KMP, but it uses a different function than $\pi$ and it has an interesting different application than string matching.
+
+Instead of using $\pi$, it uses the **z-function**.
+Given a position, this function gives the length of the longest string that's both the prefix of $S$
+and of the suffix of $S$ starting at the given position.
+
+Here's some examples of what this function might look like:
+- `aabxaayaab` $\rightarrow$ $Z=[0,1,0,0,2,1,0,3,1,0]$
```

Comment:
now you'd think z_0 would be the length of the string...

Reply:
Yes, I put 0 because first is considered unimportant. Multiple sources say the same, for example CP Algorithms:
![image](https://github.com/cpinitiative/usaco-guide/assets/59292725/c75aa300-2c3e-47d1-8097-4956cd429219)

Reply:
![image](https://github.com/cpinitiative/usaco-guide/assets/59292725/b05ee279-f952-4d12-b41d-ef6282b10515)

Reply:
CPH considers the same
![image](https://github.com/cpinitiative/usaco-guide/assets/59292725/8aed7a1e-3273-4895-bbee-fc53b90a5dd6)

Reply:
i mean why would it not be well defined though?

Reply:
ok i changed it